Codec/Archive/Tar/Read.hs

```diff
@@ -187,9 +183,14 @@ getOct off len = parseOct
     -- which is what I will assume.
     parseOct s | BS.head s == 128 = return (readBytes (BS.tail s))
                | BS.head s == 255 = return (negate (readBytes (BS.tail s)))
-    parseOct s  = case readOct s of
-      Just x  -> return x
-      Nothing -> Error HeaderBadNumericEncoding
+    parseOct s
+      | BS.null stripped = return 0
```

Comment:
Can you explain in what case this codepath is necessary? Is this specified somewhere? Why does the producing tar program not use `0x30`?

Reply:
Note that this check is inherited from the original code (line 176) so this is not a change. I'm fairly certain the check was necessary when I tested it, but I can't say any more than that.